core/services/keeper/models.go

```diff
@@ -56,9 +57,38 @@ func (k *KeeperIndexMap) Value() (driver.Value, error) {
 }
 
 func (upkeep UpkeepRegistration) PrettyID() string {
-	result, err := utils.Uint256ToBytes(upkeep.UpkeepID.ToInt())
+	return NewUpkeepIdentifier(upkeep.UpkeepID).String()
+}
+
+func NewUpkeepIdentifier(i *utils.Big) *UpkeepIdentifier {
+	val := UpkeepIdentifier(*i)
+	return &val
+}
+
+type UpkeepIdentifier utils.Big
+
+// String produces a hex encoded value, zero padded, prefixed with UpkeepPrefix
+func (ui UpkeepIdentifier) String() string {
+	val := utils.Big(ui)
+	result, err := utils.Uint256ToBytes(val.ToInt())
 	if err != nil {
 		panic(errors.Wrap(err, "invariant, invalid upkeepID"))
 	}
-	return "UPx" + hex.EncodeToString(result)
+	return fmt.Sprintf("%s%s", UpkeepPrefix, hex.EncodeToString(result))
+}
+
+// Value returns this instance serialized for database storage.
+func (ui UpkeepIdentifier) Value() (driver.Value, error) {
+	return ui.String(), nil
```

Comment:
Not sure if you are using this to write to DB already, but changing the format will require a DB migration. Probably easier to just write as a bigInt

Reply:
I added this to provide the ability to start using the `UpkeepIdentifier` type directly in the ORM. It implements the Scanner and Valuer interface. This function returns the same output as big.Int when writing to the database: a string value.

I can remove both the scanner and valuer implementations in favor of adding them to future refactors. I would highly suggest we use UpkeepIdentifier in place of utils.Big or big.Int to introduce more consistency, readability, and type safety to the code. The only way we have to identify that either of these value types would be an upkeepID is in the variable name alone which creates ambiguity and inconsistent formatting, as you pointed out in the associated story.

Reply:
I just realized that this is an error as the output will be `UPx...` which we don't want in the database. I'll fix that and add a unit test to catch the case. Even if it doesn't end up in this PR.